File: ecs/cdk/stack/ecs-stack.ts
import * as cdk from 'aws-cdk-lib';
import { Construct } from 'constructs';
import * as ec2 from 'aws-cdk-lib/aws-ec2';
import * as ecs from 'aws-cdk-lib/aws-ecs';
import * as ecsPatterns from 'aws-cdk-lib/aws-ecs-patterns';

export class EcsStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const vpc = new ec2.Vpc(this, 'Vpc', {
      maxAzs: 2,
    });

    const cluster = new ecs.Cluster(this, 'Cluster', {
      vpc,
    });

    const nginxService = new ecsPatterns.ApplicationLoadBalancedFargateService(this, 'NginxService', {
      cluster,
      desiredCount: 1,
      taskImageOptions: {
        image: ecs.ContainerImage.fromRegistry('nginx'),
        containerPort: 80,
      },
    });

    const vueService = new ecsPatterns.ApplicationLoadBalancedFargateService(this, 'VueService', {
      cluster,
      desiredCount: 1,
      taskImageOptions: {
        image: ecs.ContainerImage.fromRegistry('node:22-alpine'),
        containerPort: 8080,
      },
    });
  }
}

File: ecs/cdk-test/src/main/java/com/example/EcsStack.java
package com.example;

import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.services.ec2.Vpc;
import software.amazon.awscdk.services.ecs.Cluster;
import software.amazon.awscdk.services.ecs.patterns.ApplicationLoadBalancedFargateService;
import software.amazon.awscdk.services.ecs.patterns.ApplicationLoadBalancedTaskImageOptions;
import software.amazon.awscdk.services.ecs.ContainerImage;

public class EcsStack extends Stack {
    public EcsStack(final App scope, final String id, final StackProps props) {
        super(scope, id, props);

        Vpc vpc = Vpc.Builder.create(this, "Vpc").maxAzs(2).build();

        Cluster cluster = Cluster.Builder.create(this, "Cluster")
                .vpc(vpc)
                .build();

        ApplicationLoadBalancedFargateService nginxService = ApplicationLoadBalancedFargateService.Builder.create(this, "NginxService")
                .cluster(cluster)
                .desiredCount(1)
                .taskImageOptions(ApplicationLoadBalancedTaskImageOptions.builder()
                        .image(ContainerImage.fromRegistry("nginx"))
                        .containerPort(80)
                        .build())
                .build();

        ApplicationLoadBalancedFargateService vueService = ApplicationLoadBalancedFargateService.Builder.create(this, "VueService")
                .cluster(cluster)
                .desiredCount(1)
                .taskImageOptions(ApplicationLoadBalancedTaskImageOptions.builder()
                        .image(ContainerImage.fromRegistry("node:16-alpine"))
                        .containerPort(8080)
                        .build())
                .build();
    }
}